bids/layout/index.py

```diff
@@ -164,10 +165,37 @@ def index_files(self):
         """Index all files in the BIDS dataset. """
         self._index_dir(self.root, self.config)
 
-    def index_metadata(self):
+    def index_metadata(self, **filters):
         """Index metadata for all files in the BIDS dataset. """
+        if filters:
+            default_ext = ['[.]+']
+            # ensure we are returning objects
+            filters['return_type'] = 'object'
+            # until 0.11.0, user can specify extension or extensions
+            if filters.get('extension'):
+                ext_key = 'extension'
+            elif filters.get('extensions'):
+                ext_key = 'extensions'
+            else:
+                ext_key = 'extension'
```

Comment:
```suggestion

```

(Sorry apparently its really cumbersome to make a multi-line suggestion, or I'm missing something)